### DISCUSSION TITLE
shrink event emitting

### DISCUSSION THREAD

**Discussion by Haroenv** [algolia/instantsearch#5223] [Ideas]
We're currently relying on `@algolia/events` (a fork of `events` to ensure the node version isn't taken), but it's a little large (8kb uncompressed) due to its large api surface and it not being builtin. I can see a couple ideas:

- no longer inherit from EventEmitter (breaking change)
- simplify largely what's inside EventEmitter (breaking change, not if it's not inherited)

<details><summary>what could be inside</summary>

```js

function EventEmitter() {
  this._events = this._events || {};
}
module.exports = EventEmitter;

EventEmitter.prototype._events = {};

EventEmitter.prototype.emit = function (type) {
  var er, handler, len, args, i, listeners;

  // If there is no 'error' event listener then throw.
  if (type === 'error') {
    if (
      !this._events.error ||
      (this._events.error && !this._events.error.length)
    ) {
      er = arguments[1];
      if (er instanceof Error) {
        throw er; // Unhandled 'error' event
      } else {
        // At least give some kind of context to the user
        var err = new Error(
          'Uncaught, unspecified "error" event. (' + er + ')'
        );
        err.context = er;
        throw err;
      }
    }
  }

  handler = this._events[type];

  if (!handler) return false;

  args = Array.prototype.slice.call(arguments, 1);
  listeners = handler.slice();
  len = listeners.length;
  for (i = 0; i < len; i++) listeners[i].apply(this, args);

  return true;
};

EventEmitter.prototype.addListener = function (type, listener) {
  if (!this._events[type]) this._events[type] = [listener];
  // If we've already got an array, just append.
  else this._events[type].push(listener);

  return this;
};

EventEmitter.prototype.on = EventEmitter.prototype.addListener;

EventEmitter.prototype.removeListener = function (type, listener) {
  this._events[type] = this._events[type].filter(function (l) {
    return l !== listener;
  });

  return this;
};

EventEmitter.prototype.removeAllListeners = function () {
  this._events = {};

  return this;
};
```

</details>

- use the builtin EventTarget (breaking change)
  - node now also supports EventTarget, meaning we wouldn't need a dependency for this anymore? needs to be investigated!

**Comment by qndel** [comment]
![image](https://user-images.githubusercontent.com/14297035/209916808-4827702a-9071-4232-b32a-ddb231d521b4.png)
what is this sorcery? :O

Okay figured it out - looks like there was some spambot/scambot with "hurr durr check my pics here" that tagged a lot of people then ... deleted the message? Even discord bot didn't pick it up, only email notification :D Sorry for the confusion

**Reply by Haroenv** [reply]
That's weird, I guess you can report it to GitHub, as I indeed didn't mention you
